src/lib/support/Span.h

```diff
@@ -38,6 +41,11 @@ class Span
 
     const T * data() const { return mDataBuf; }
     size_t size() const { return mDataLen; }
+    bool empty() const { return size() == 0; }
+    bool data_equal(const Span & other) const
+    {
+        return !empty() && !other.empty() && (size() == other.size()) && (memcmp(data(), other.data(), size() * sizeof(T)) == 0);
```

Comment:
So this will return false for two empty spans, right?  But shouldn't those test true for `data_equal`?

Reply:
Yes, you are probably right. Update to perform comparison for zero data as well